haskell-api.md

```diff
@@ -117,6 +117,107 @@ possible. Issues include:
 If you are creating a new `PUT` consider if it could be expressed in a more
 granular `PATCH` semantic.
 
+### `PATCH` semantics
+
+Modifications to an existing resource should be made via a `PATCH` request.
+
+### Specific guidelines
+
+A well-formed `PATCH` endpoint
+
+- Should update an entity's `updatedAt` field, if present.
+- Should return an empty response or the modified resource. Ideally, if the
+  latter, the resource should be re-fetched for ease of testing and to ensure
+  the update was correctly persisted.
+- Should accept `null` to unset nullable fields, disallowing it for
+  non-nullables.
+- Should 404 if a resource with the route's identifier(s) does not exist.
+- Should not allow for update-scoping information within the `PATCH` object. If
+  scoping information is absolutely necessary, it should be include in query
+  params.
```

Comment:
I'm not sure what this means, maybe a `Good/Bad` code listing would be useful?

Reply:
Yeah, this one is tricky to talk about!

It's summarizing a pattern I undid in our code where we had a request like
```
PUT /2/courses/1234/enabledDashboardModes
{
  modes: [...],
  grade: null|number
}
```

`grade` would scope the request to only modify for certain grades. So this would be suggesting
```
PATCH /2/courses/1234/enabledDashboardModes?[grade=number]
{
  modes: [...]
}
```
instead.

Does that make sense? Perhaps this is not worth mentioning.

Reply:
I agree that is an odd behavior, and would be good to avoid. I'm good to keep it as long as there's an example shown. I'm also fine if you want to leave it off for now.